### PR TITLE
fix #85 :  Reorder generic types on createStore function

### LINKED ISSUE #85: Permit to specify the State structure when using createStore
## Feature request
_[According to this twitter discussion](https://github.com/posva/pinia/blob/master/src/store.ts#L164)_
### Need
In order to getting more solid store typing, it could be nice to specify the type which will reflect the store. Currently, the state is inferred by the state initialization.
To be honest, it's already possible by this way : 
```ts
type MainStare = {
  counter: number;
  name: string;
}

export const useMainStore = createStore<string, MainState>({
  id: 'main',
  state: () => ({
    counter: 0,
    name: 'Edimitchel',
  }),
  // ...
})
```
As you can see, the current API forced us to specify the first generic type ([see here](https://github.com/posva/pinia/blob/master/src/store.ts#L164)).

### My request
**My simple request is following :** Could it be possible to swap [these two lines](https://github.com/posva/pinia/blob/master/src/store.ts#L163-L164) ?
_If accepted, I would create the PR_

The result would be like this (with a better usage if we have nested or optional state) : 

```ts
export type MainStare = {
  counter: number;
  name: string;
  team?: {
    name: string;
    createdAt?: Date;
  }
}

export const useMainStore = createStore<MainState>({
  id: 'main',
  state: () => ({
    counter: 0,
    name: 'Edimitchel',
    team: {
      name: 'Vue Community'
    }
  }),
  // ...
})
```
_______
I guess, following could be subject to another issue by amending the `createStore` signature :

```ts
export type MainStare = {
  counter: number;
  name: string;
  team?: {
    name: string;
    createdAt?: Date;
  }
}

export const useMainStore = createStore<MainState>('main', {
  state: () => ({
    counter: 0,
    name: 'Edimitchel',
    team: {
      name: 'Vue Community'
    }
  }),
  // ...
})
```

### PR DESCRIPTION
**Pull request by edimitchel**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [ ] Making state more solid to ensure type safe !

See #85

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -160,10 +160,10 @@ export function buildStore<
  * @param options
  */
 export function createStore<
-  Id extends string,
   S extends StateTree,
-  G extends Record<string, StoreGetter<S>>,
-  A extends Record<string, StoreAction>
+  Id extends string = string,
```
Comment (posva) [inline src/store.ts]: let's move it to the end

File: src/store.ts [hunk 2]
```diff
@@ -160,10 +160,10 @@ export function buildStore<
...
+  G extends Record<string, StoreGetter<S>> = Record<string, StoreGetter<S>>,
+  A extends Record<string, StoreAction> = Record<string, StoreAction>
```
Comment (posva) [inline src/store.ts:166]:
```suggestion
  G extends Record<string, StoreGetter<S>>,
  A extends Record<string, StoreAction>,
  Id extends string
```

Comment (posva) [reply src/store.ts:166]: What I don't really like about the defaults is that you loose all typing if you only provide the State type in the generic `createStore<State>()`

I think it makes more sense to manually type the return of the function like said at https://github.com/posva/pinia/issues/85#issuecomment-631473019

Comment (edimitchel) [reply src/store.ts:166]: Yes, it's a second way to do. But personally, I prefer to use generic feature and not polluating the body of my function.

**Pull request closed without merging**